source/javascripts/app/_search.js

```diff
@@ -27,24 +29,46 @@
         body: body.text()
       });
     });
+
+    determine_search_delay();
+  }
+  function determine_search_delay() {
+      if(index.tokenStore.length>5000) {
+        search_delay = 300;
+      }
   }
 
+  var wait = function() {
+    return function(executing_function, wait_time){
+          clearTimeout (timeout_handle);
+          timeout_handle = setTimeout(executing_function, wait_time);
+        };
```

Comment:
Mind cleaning up the indentation here a bit, and throughout the changes? Looks like we're mixing 4 and 2 spaces, it should probably just be 2 (I think?)